### DISCUSSION TITLE
How to provide generic for write(action) atom agruments?

### DISCUSSION THREAD

**Discussion by geril07** [pmndrs/jotai#2175] [Q&A] [answered]
```
type SomeType = "a" | "b"

const someAtom = atom(null, (get,set, someValue : T) => {}) // How to set kinda <T extends SomeType>(get,set, someValue: T) => {} ?
```
I want to provide generic for `someValue` where T expected to be `<T extends SomeType>`.

Can i do this?

**Comment by dai-shi** [comment, marked as answer]
I don't think it's possible in general, because `someAtom` is not a function, but an object.

Here's a possible workaround: https://tsplay.dev/w1Q4Ow

<img width="763" alt="image" src="https://github.com/pmndrs/jotai/assets/490574/3c5a301c-2e0d-4e76-9087-aac9ec8eabee">

**Reply by geril07** [reply]
In this workaround and atom structure, unfortunately generics(i provide simple example but i want to depend on another argument) goes away. For me the solution was to write just plain function which take `get/set`(if needed), and other arguments. And just calling it inside of atom with `someWriter(get, someValue)`. For now i don't need to use it inside of components(only in other atoms), so seems fine for me.
